share/p2p/peers/manager.go

```diff
@@ -278,18 +274,16 @@ func (s *Manager) hashIsBlacklisted(hash share.DataHash) bool {
 	return s.blacklistedHashes[hash.String()]
 }
 
-func (p *syncPool) markValidated() {
-	p.isValidatedDataHash.Store(true)
-}
-
 func (s *Manager) GC(ctx context.Context) {
 	ticker := time.NewTicker(gcInterval)
 	defer ticker.Stop()
 
 	var blacklist []peer.ID
 	for {
 		blacklist = s.cleanUp()
-		s.blacklistPeers(blacklist...)
+		if len(blacklist) > 0 {
+			s.blacklistPeers(blacklist...)
+		}
```

Comment:
do you need this check? It'll just short circuit as there's nothing to loop over

Reply:
It was producing misleading and scary logs of "blacklisting peers, peers:[]" would rather just skip :)